src/run/ongoing.ts

```diff
@@ -0,0 +1,37 @@
+import { ApiClient } from "../client/apiClient";
+import { StartedRun } from "./start";
+import { Assertion, RunInformationResponse } from "../client/responses/runInformationResponse";
+import { setTimeout } from "timers/promises";
+import * as core from "@actions/core";
+import { isRunning, statusName } from "./status";
+
+export interface FinishedRun {
+  runId: String;
+  statusCode: number;
+  statusName: string;
+  assertions: Assertion[];
+}
+
+export const waitForRunEnd = async (client: ApiClient, startedRun: StartedRun): Promise<FinishedRun> => {
+  await setTimeout(1000); // add 1s delay because run duration might not be populated yet
+  let runInfo: RunInformationResponse;
+  let oldStatus: number = -1;
+  while (true) {
```

Comment:
```suggestion
  while (isRunning(runInfo?.status ?? RUNNING) {
```

where RUNNING is a running value (for he first time)
Or a do `while construct`?

My point: I want to avoid `while(true)` at all cost